Description: remove DefaultDependencies from cloud-init-hotplug.socket (#5722)
 Shifting systemd ordering earlier in boot for cloud-init-hotplugd.socket
 introduced a systemd ordering cycle due to After=sysinit.target being pulled
 carried along witu cloud-init-hotplugd.socket in earlier boot. This caused
 conflicts in only Ubuntu Live Desktop images which have a custom dropin from
 livecd-rootfs which also orders cloud-init-network.service
 After=NetworkManager.service. The livecd-rootfs override is documented in
 LP: #2081325. This bug is not seen on Ubuntu Server images.
Author: Chad Smith <chad.smith@canonical.com>
Origin: upstream
Bug: https://bugs.launchpad.net/ubuntu/+source/cloud-init/+bug/2081124
Last-Update: 2024-09-20
--- a/systemd/cloud-init-hotplugd.socket
+++ b/systemd/cloud-init-hotplugd.socket
@@ -5,6 +5,9 @@
 # Known bug with an enforcing SELinux policy: LP: #1936229
 [Unit]
 Description=cloud-init hotplug hook socket
+DefaultDependencies=no
+Before=shutdown.target
+Conflicts=shutdown.target
 After=cloud-config.target
 ConditionPathExists=!/etc/cloud/cloud-init.disabled
 ConditionKernelCommandLine=!cloud-init=disabled
